agent_build/package_builder/builder_inside_docker/__init__.py

```diff
@@ -0,0 +1,108 @@
+import pathlib as pl
+import abc
+import sys
+import shlex
+from typing import List, Union
+
+from agent_build.docker_images.image_builder import ImageBuilder
+from agent_build.package_builder import builder
+from agent_build import common
+
+__all__ = ["PackageBuilderInsideDocker"]
+
+_PARENT_DIR = pl.Path(__file__).parent
+
+
+class PackageBuilderInsideDocker(builder.PackageBuilder, metaclass=abc.ABCMeta):
+    BASE_IMAGE: ImageBuilder = None
+    DOCKERFILE_PATH: Union[str, pl.Path] = _PARENT_DIR / "Dockerfile"
+
+    @classmethod
+    def _get_used_files(cls):
+        return cls.BASE_IMAGE.get_used_files()
+
+    @classmethod
+    def prepare_dependencies(
+            cls,
+            cache_dir:
+            Union[str, pl.Path] = None,
+    ):
+        cls.BASE_IMAGE.build(cache_dir=cache_dir)
+
+    def _build_in_docker(
+            self,
+            build_info: common.PackageBuildInfo,
+            output_path: Union[str, pl.Path],
+    ):
+        # if we build package inside the docker, then we need to pre-build the base image first.
+        self.prepare_dependencies()
+
+        # Build the package using docker.
+        image_name = f"scalyr-agent-{self.package_type_name}-package-builder"
+
+        # When package is build inside the docker, the trick is to run the script 'agent_build/build_package.py
+        # once more but inside the docker image build.
+        new_argv = []
+        i = 0
+        while True:
```

Comment:
Add comments for what this while loop is doing.

Reply:
This code part has also been removed. The simplest(i hope) version of that had moved to the agent_build/build_package.py file.